pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -124,14 +125,14 @@ func (o createAuthInfoOptions) run() error {
 }
 
 // authInfo builds an AuthInfo object from the options
-func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
+func (o *createAuthInfoOptions) modifyAuthInfo(cmd *cobra.Command, existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
-		certPath := o.clientCertificate.Value()
-		if o.embedCertData.Value() {
+	if cmd.Flags().Changed(clientcmd.FlagCertFile) {
```

Comment:
Thing is, this destroys the ability to reuse this logic in composable ways as part of different commands. Before, you could take the command options struct, bind different flags to specific fields, set others programmatically, and reuse the execute logic. Now, the execute logic cares about the command line flag string that populated its field. It's coupling the wrong things.